### ISSUE TITLE
Support lzma (xz) compression

### ISSUE THREAD

**Issue by Nikratio** [archiecobbs/s3backer#148]
Would it be feasible for s3backer to support alternative compression methods?

At least in a residential setting, even with a very expensive compression algorithm it is likely that the bottleneck is network bandwidth rather than CPU time for compression/decompression. Therefore, supporting an algorithm like LZMA would - at least for me - pay off.

**Comment by archiecobbs** [comment]
Yes this is feasible and not too hard once one is familiarized with the LZMA C API.

**Comment by archiecobbs** [comment]
FYI in commit 2214e091 the code is refactored to support more than one compression algorithm.
